ucp/benchmarks/README.md

```diff
@@ -0,0 +1,117 @@
+
+The cuDF merge benchmark can be executed in 3 different ways:
+
+1. Local-node only;
+1. Multi-node with automatic process launch via SSH;
+1. Multi-node with manual process launch.
+
+In the following subsections we will exemplify how to launch the benchmark for each case. Be sure to check `python -m ucp.benchmarks.cudf_merge --help` to see a complete description of available options and their description.
+
+
+### Local-node only
+
+This is the simplest setup and will execute the benchmark only on GPUs where the process is being launched from. In its simple form, it can be executed as:
+
+```bash
+python -m ucp.benchmarks.cudf_merge -d 0,1 -c 1_000_000 --iter 10
+```
+
+The process above will execute the benchmark with the first 2 GPUs in the node, with 1M rows per chunk and a single chunk per node, for 10 iterations. To extend the same to 8 GPUs we could simply write:
+
+```bash
+python -m ucp.benchmarks.cudf_merge -d 0,1,2,3,4,5,6,7 -c 1_000_000 --iter 10
+```
+
+
+### Multi-node with automatic process launch via SSH
+
+In this setup, the user can run the benchmark spanning multiple nodes, but a password-less SSH setup is required. That is often achieved by having a private key that doesn't require a passphrase authorized to login to all the nodes where the benchmark is expected to run. To test if this setup is already correct, it should suffice to execute `ssh NODE_HOSTNAME` or `ssh NODE_IP`, if you get a shell in the remote machine without being asked for a password, you are probably good to go. If this is not yet setup, please consult the documentation of your operating systems to set this up.
+
+Once the SSH setup is complete, we can extend the local-node example by simply introducing a new argument `--hosts`. The `--hosts` argument takes a comma-separated list of node hostnames or IP addresses, where the first element is where the server will run, followed by any number of worker hosts. The server is required to synchronize workers and doesn't require a GPU but is allowed to share the same node as one of the worker nodes in that list. For example, if we want the server to run on a node with hostname 'dgx12', with workers on machine with hostname 'dgx12', another with IP address '10.10.10.10' and another with hostname 'dgx13', we would run the command as below:
+
+```bash
+python -m ucp.benchmarks.cudf_merge -d 0,1,2,3 -c 1_000_000 --iter 10 --hosts dgx12,dgx12,10.10.10.10,dgx15
+```
+
+Note that in the example above we repeated dgx12, the first node identifies the server, and the remaining arguments identify workers. Multi-node setups also require every node to use the same GPU indices, so if you specify `-d 0,1,2,3`, all worker nodes must have those 4 GPUs available.
+
+Should anything go wrong, you may specify the `UCXPY_ASYNCSSH_LOG_LEVEL=DEBUG` environment variable. This will print additional information, including the exact command that is being executed in server and workers, as well as the output for each process.
```

Comment:
```suggestion
Should anything go wrong, you may specify the `UCXPY_ASYNCSSH_LOG_LEVEL=DEBUG` environment variable. This will print additional information, including the exact command that is being executed on the server and workers, as well as the output for each process.
```